### PR TITLE
Pioneer DDJ-400: make Beat FX section more intuitive

### PR DESCRIPTION
**Pull request by tiesjan**
This patch makes the mapping of the Beat FX section on the Pioneer DDJ-400 slightly more intuitive:
- `Beat <` (LEFT) and `Beat >` (RIGHT) buttons now cycle through the focus of the three Effects in Effect Unit 1 leftward and rightward, respectively.
- `FX SELECT v` (DOWN) and `FX SELECT v` `+SHIFT` (UP) buttons now select the next and previous effect entry for the focused Effect, respectively.

This is more in line with the controller's button names and the Mixxx interface.

Has been tested on my own controller (with the latest firmware) using Mixxx 2.3 to work as expected.

Based on the work proposed in #4662. Changes to the Mixxx manual proposed in mixxxdj/manual#519.

### CODE REVIEW

**Comment by tiesjan** [comment]
@ronso0 @daschuer This was originally planned for 2.3.4, but (understandably) left out as a non-blocking issue. Would it be possible to add it to 2.3.5? That would be much appreciated. 🙂

File: res/controllers/Pioneer-DDJ-400.midi.xml [hunk 1]
```diff
@@ -1028,23 +1018,23 @@
...
+
             <control>
-                <description>BEAT RIGHT + shift - select next effect for the current unit</description>
+                <description>BEAT FX SELECT - press - load next effect entry for focused Effect</description>
                 <group>[EffectRack1_EffectUnit1]</group>
-                <key>PioneerDDJ400.beatFxSelectNextEffect</key>
+                <key>PioneerDDJ400.beatFxSelectPressed</key>
                 <status>0x94</status>
-                <midino>0x6B</midino>
+                <midino>0x63</midino>
```
Comment (Swiftb0y) [inline res/controllers/Pioneer-DDJ-400.midi.xml:1027]: what this a bug in the previous version or whats the rationale for this change.

**Review by Swiftb0y — COMMENTED** [review]
Changes mostly make sense. Sorry for neglecting this PR for so long.

File: res/controllers/Pioneer-DDJ-400-script.js [hunk 2]
```diff
@@ -243,30 +242,44 @@ PioneerDDJ400.beatFxLevelDepthRotate = function(_channel, _control, value) {
...
-    engine.setValue("[EffectRack1_EffectUnit1]", "focused_effect", 1);
+    var focusedEffect = engine.getValue("[EffectRack1_EffectUnit1]", "focused_effect");
+
+    if (focusedEffect <= 1) {
+        focusedEffect = 3;
+    } else {
+        focusedEffect -= 1;
+    }
+
+    engine.setValue("[EffectRack1_EffectUnit1]", "focused_effect", focusedEffect);
```
Comment (Swiftb0y) [inline res/controllers/Pioneer-DDJ-400-script.js:256]: You seem to be competent in javascript. Could you deduplicate this a bit by factoring out the common parts into a function like this:
```javascript
PioneerDDJ400.changeFocusedEffectBy = function (numberOfSteps) {
        var focusedEffect = engine.getValue("[EffectRack1_EffectUnit1]", "focused_effect");
        var numberOfEffectsPerEffectUnit = 3;
        // standard euclidian modulo by use of two plain modulos
        focusedEffect = (((focusedEffect + numberOfSteps) % numberOfEffectsPerEffectUnit) + numberOfEffectsPerEffectUnit ) % numberOfEffectsPerEffectUnit;
        engine.setValue("[EffectRack1_EffectUnit1]", "focused_effect", focusedEffect);
}
```
Then use that function here (and `PioneerDDJ400.changeFocusedEffectBy(1);` for the right button). 
```suggestion
    PioneerDDJ400.changeFocusedEffectBy(-1);
```

Code cleanups like this usually help us developers tremendously to reduce the amount of time needed to review a mapping and also makes it a much more pleasant experience, which results in mappings being reviewed earlier and faster. Thank you.

Comment (tiesjan) [reply res/controllers/Pioneer-DDJ-400-script.js]: Thanks for the suggestion. I've had to take the one-based indexes of the effects into account, but seems to be cleaner now.

Comment (Swiftb0y) [reply res/controllers/Pioneer-DDJ-400-script.js]: Thank you, I didn't notice that.

**Review by Swiftb0y — APPROVED** [review]
Thank you, waiting for CI

**Comment by tiesjan** [comment]
Great! Force-pushed to fix the [code style issues](https://github.com/mixxxdj/mixxx/actions/runs/4470844691/jobs/7854967789).

**Comment by Swiftb0y** [comment]
Thank you

**Pull request merged**